tests/publisher/tests_register_name.py

```diff
@@ -254,37 +124,21 @@ def test_name_reserved(self):
 
         response = self.client.post(self.endpoint_url, data=self.data)
 
-        assert response.status_code == 302
-        self.assertIn("snap_name=test-snap", response.location)
-        self.assertIn("is_private=False", response.location)
-        self.assertIn("/register-snap", response.location)
+        assert response.status_code == 200
 
     @responses.activate
     def test_claim_dispute(self):
-        payload = {"error_list": [{"code": "already_claimed"}]}
-        responses.add(responses.POST, self.api_url, json=payload, status=409)
-
-        user_payload = {"error_list": [], "stores": []}
-        responses.add(
-            responses.GET, self.user_url, json=user_payload, status=200
-        )
-
-        response = self.client.post(self.endpoint_url, data=self.data)
-
-        assert response.status_code == 302
-        self.assertEqual(response.location, "/admin/account")
-
-    @responses.activate
-    def test_post_error_user_error(self):
-        payload = {"error_list": [{"code": "oops"}]}
-        responses.add(responses.POST, self.api_url, json=payload, status=409)
-
-        user_payload = {"error_list": [], "stores": []}
-        responses.add(
-            responses.GET, self.user_url, json=user_payload, status=502
-        )
+        payload = {
+            "success": False,
+            "data": {
+                "snap_name": "test-snap",
+                "is_private": "private",
+                "store": "store",
+                "error_code": "already_claimed",
+            },
+        }
+        responses.add(responses.POST, self.api_url, json=payload, status=200)
 
         response = self.client.post(self.endpoint_url, data=self.data)
 
-        assert response.status_code == 502
-        self.assert_template_used("50X.html")
+        assert response.status_code == 200
```

Comment:
Should it assert for anything else about response? Right now it only seems to check the status code?

Reply:
I've added a test for the response data